salt/modules/salt_version.py

```diff
@@ -0,0 +1,176 @@
+# -*- coding: utf-8 -*-
+'''
+Access Salt's elemental release code-names.
+
+.. versionadded:: Neon
+
+Salt's feature release schedule is based on the Periodic Table, as described
+in the :ref:`Version Numbers <version-numbers>` documentation.
+
+Since deprecation notices often use the elemental release code-name when warning
+users about deprecated changes, it can be difficult to build out future-proof
+functionality that are dependent on a naming scheme that moves.
+
+For example, a state syntax needs to change to support an option that will be
+removed in the future, but there are many Minion versions in use across an
+infrastructure. It would be handy to use some Jinja syntax to check for these
+instances to perform one state syntax over another.
+
+A simple example might be something like the following:
+
+.. code-block:: jinja
+
+    {# a boolean check #}
+    {% set option_deprecated = salt['salt_version.less_than']("Sodium") %}
+
+    {% if option_deprecated %}
+      <use old syntax>
+    {% else %}
+      <use new syntax>
+    {% endif %}
+
+'''
+
+# Import Python libs
+from __future__ import absolute_import, print_function, unicode_literals
+import logging
+
+# Import Salt libs
+from salt.ext import six
+import salt.version
+import salt.utils.versions
+
+
+log = logging.getLogger(__name__)
+
+__virtualname__ = 'salt_version'
+
+
+def __virtual__():
+    '''
+    Only work on POSIX-like systems
+    '''
+    return __virtualname__
+
+
+def get_release_number(name):
+    '''
+    Returns the release number of a given release code name in a
+    ``<year>.<month>`` context.
```

Comment:
Probably should just change this to `Returns the release number of a given codename`.

Reply:
I changed it to

> Returns the release number of a given release codename in a ``MAJOR.PATCH`` format.

in line with the SEP:

> With the `neon` release, to indicate the new change in release process, Salt will change to a new, non-date based version schema beginning at 3000. The version will be MAJOR.PATCH. For a planned release containing features and/or bug fixes the MAJOR version will be incremented